ansible/configs/open-training-novelloshell/pre_infra.yml

```diff
@@ -0,0 +1,20 @@
+---
+- name: Step 000 Pre Infrastructure
+  hosts: localhost
+  connection: local
+  become: false
+  tags:
+  - step001
+  - pre_infrastructure
+  - osp_migration
+  tasks:
+  - name: Create migration host group
+    add_host:
+      name: "{{ import_host }}"
+      ansible_become: true
+      ansible_ssh_private_key_file: "{{ migration_key_path | default(omit) }}"
```

Comment:
You should not need a static key. But rather generate one and add the public part as authorized_keys?
or grab the key from the heat template output?
Please have a look at https://redhat-cop.github.io/agnosticd/#_ssh_keys_and_access
There are roles available in agnosticd to help you with the creation, with some conventions regarding the names.